Source/Moq.Analyzers/Helpers.cs

```diff
@@ -48,12 +40,12 @@ internal static bool IsCallbackOrReturnInvocation(SemanticModel semanticModel, I
         };
     }
 
-    internal static InvocationExpressionSyntax? FindSetupMethodFromCallbackInvocation(SemanticModel semanticModel, ExpressionSyntax expression)
+    internal static InvocationExpressionSyntax? FindSetupMethodFromCallbackInvocation(SemanticModel semanticModel, ExpressionSyntax expression, CancellationToken cancellationToken)
     {
         InvocationExpressionSyntax? invocation = expression as InvocationExpressionSyntax;
         if (invocation?.Expression is not MemberAccessExpressionSyntax method) return null;
-        if (IsMoqSetupMethod(semanticModel, method)) return invocation;
-        return FindSetupMethodFromCallbackInvocation(semanticModel, method.Expression);
+        if (IsMoqSetupMethod(semanticModel, method, cancellationToken)) return invocation;
```

Comment:
So that I don't work at cross-purposes, what do you want to change it to?

Currently I've been rewriting this

```csharp
if (IsMoqSetupMethod(semanticModel, method, cancellationToken)) return invocation;
```

as this

```csharp
if (IsMoqSetupMethod(semanticModel, method, cancellationToken))
{
    return invocation;
}
```

which is generally my preference, but gets a bit annoying in Roslyn code.

I'd be OK with allowing this as well

```csharp
if (IsMoqSetupMethod(semanticModel, method, cancellationToken)) { return invocation; }
```

However that currently triggers SA1501.

Interestingly this

```csharp
if (IsMoqSetupMethod(semanticModel, method, cancellationToken)) return invocation;
```

is currently fine, though in VS it looks like we have SA1503 on, which I think _should_ trigger here.

Reply:
So if I had my way I'd turn off SA1501 and figure out why SA1503 isn't firing. Is that what you expect as well?

Reply:
The brace on the new line gets annoying because there's so many if checks that are necessary to qualify the rule?

Reply:
@MattKotsenas and I spoke in another forum. Agreed that what we have is currently fine and that where possible we should standardize on that (e.g., defensive parameter checks and these Roslyn checks) to avoid triangle code.